exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,2055 @@
+	if arg.OrderType != OkxOrderMarket && arg.OrderType != OkxOrderLimit && arg.OrderType != OkxOrderPostOnly &&
+		arg.OrderType != OkxOrderFOK && arg.OrderType != OkxOrderIOC && arg.OrderType != OkxOrderOptimalLimitIOC {
+		return nil, errInvalidOrderType
+	}
+	if arg.Amount <= 0 {
+		return nil, errInvalidAmount
+	}
+	if arg.Price <= 0 && (arg.OrderType == OkxOrderLimit || arg.OrderType == OkxOrderPostOnly ||
+		arg.OrderType == OkxOrderFOK || arg.OrderType == OkxOrderIOC) {
+		return nil, fmt.Errorf("invalid order price for %s order types", arg.OrderType)
+	}
+	if arg.QuantityType != "base_ccy" && arg.QuantityType != "quote_ccy" {
+		arg.QuantityType = ""
+	}
+	randomID, err := common.GenerateRandomString(32, common.SmallLetters, common.CapitalLetters, common.NumberCharacters)
+	if err != nil {
+		return nil, err
+	}
+	input := WsPlaceOrderInput{
+		ID:        randomID,
+		Arguments: []PlaceOrderRequestParam{*arg},
+		Operation: okxOpOrder,
+	}
+	err = ok.Websocket.AuthConn.SendJSONMessage(input)
+	if err != nil {
+		return nil, err
+	}
+	timer := time.NewTimer(ok.WebsocketResponseMaxLimit)
+	wsResponse := make(chan *wsIncomingData)
+	ok.WsResponseMultiplexer.Register <- &wsRequestInfo{
+		ID:   randomID,
+		Chan: wsResponse,
+	}
+	defer func() { ok.WsResponseMultiplexer.Unregister <- randomID }()
+	for {
+		select {
+		case data := <-wsResponse:
+			if data.Operation == okxOpOrder && data.ID == input.ID {
+				if data.Code == "0" || data.Code == "1" {
+					resp, err := data.copyToPlaceOrderResponse()
+					if err != nil {
+						return nil, err
+					}
+					if len(resp.Data) != 1 {
+						return nil, errNoValidResponseFromServer
+					}
+					if data.Code == "1" {
+						return nil, fmt.Errorf("error code:%s message: %s", resp.Data[0].SCode, resp.Data[0].SMessage)
+					}
+					return &resp.Data[0], nil
+				}
+				return nil, fmt.Errorf("error code:%s message: %v", data.Code, ErrorCodes[data.Code])
+			}
+			continue
+		case <-timer.C:
+			timer.Stop()
+			return nil, fmt.Errorf("%s websocket connection: timeout waiting for response with an operation: %v",
+				ok.Name,
+				input.Operation)
+		}
+	}
+}
+
+// WsPlaceMultipleOrder creates an order through the websocket stream.
+func (ok *Okx) WsPlaceMultipleOrder(args []PlaceOrderRequestParam) ([]OrderData, error) {
+	for x := range args {
+		arg := args[x]
+		if arg.InstrumentID == "" {
+			return nil, errMissingInstrumentID
+		}
+		if TradeModeCross != arg.TradeMode &&
+			TradeModeIsolated != arg.TradeMode &&
+			TradeModeCash != arg.TradeMode {
+			return nil, errInvalidTradeModeValue
+		}
+		arg.Side = strings.ToLower(arg.Side)
+		if arg.Side != "buy" && arg.Side != "sell" {
+			return nil, errInvalidOrderSide
+		}
+		arg.OrderType = strings.ToLower(arg.OrderType)
+		if arg.OrderType != OkxOrderMarket && arg.OrderType != OkxOrderLimit && arg.OrderType != OkxOrderPostOnly &&
+			arg.OrderType != OkxOrderFOK && arg.OrderType != OkxOrderIOC && arg.OrderType == OkxOrderOptimalLimitIOC {
```

Comment:
Please review your changed statement here. It looks like it should have `arg.OrderType != OkxOrderOptimalLimitIOC`
Otherwise you can simplify this if statement to just `arg.OrderType == OkxOrderOptimalLimitIOC`